### ISSUE TITLE
ContainerRegistry.remove* return codes

### ISSUE THREAD

**Issue by glensc** [jdalrymple/gitbeaker#3421]
**Description**

- Node.js version: irrelevant
- Gitbeaker version: 39.17.0
- Gitbeaker release (cli, node, browser, core, requester-utils): rest
- OS & version: irrelevant

The `remove*` methods actually return gitlab http status code, 202 or 200

**Steps to reproduce**

```ts
   // code is 202
    const code = await gitlab.ContainerRegistry.removeRepository(projectId, repositoryId);

   // code is 200
    const code = await gitlab.ContainerRegistry.removeTag(projectId, repositoryId, tagName);

   // code is 202
    const code = await gitlab.ContainerRegistry.removeTags(projectId, repositoryId, nameRegexDelete);
```

**Expected behaviour**

change return type to `number`, even if it's not very useful, I would like to still test the code succeeds, i.e. checking return value is 202 or 200.

**Actual behaviour**

type script typing currently specifics `void`

**Possible fixes**

hack the type:

```ts
const code = res as any as number;
```

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [ ] I have read the documentation.

**Comment by jdalrymple** [comment]
Its a small fix! ill get around to it this week

**Comment by glensc** [comment]
> - [ ] I have read the documentation.

I have not found any other documentation than this:
- https://www.npmjs.com/package/@gitbeaker/rest

and it's rather minimal, not documenting each method arguments and return values

**Comment by glensc** [comment]
gitlab api documentation mentions 202 code:
- https://docs.gitlab.com/ee/api/container_registry.html#delete-registry-repository-tags-in-bulk

**Comment by jdalrymple** [comment]
100p, the API tries to follow the official docs as much as possible, and to avoid two sets of docs, its advised to use the official docs as the reference point.

The docs unfortunately don't mention anything about the return body, hence nothing being returned in the library. To see the status of the response, just include an optional parameter `showExpanded` to your list of arguments. The return object will then be in the form of :

```js
{
  data,
  status,
  headers,
}
```

I notice this piece of information is missing from the docs as well ( I seemed to have included it for the pagination info, but not generally). I will fix this.

**Comment by jdalrymple** [comment]
Again, for an example:

```js
 const { status } = await gitlab.ContainerRegistry.removeRepository(projectId, repositoryId, { showExpanded: true });

 const { status } = await gitlab.ContainerRegistry.removeTag(projectId, repositoryId, tagName, { showExpanded: true });

 const { status } = await gitlab.ContainerRegistry.removeTags(projectId, repositoryId, nameRegexDelete, { showExpanded: true });
    ```

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`39.18.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/39.18.0) :rocket:
